fix(learn): rework Rosetta Code Hailstone Sequence (#41279)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/hailstone-sequence.md

The Hailstone sequence of numbers can be generated from a starting positive integer, `n` by:

- If `n` is `1` then the sequence ends
- If `n` is `even` then the next `n` of the sequence `= n/2`
- If `n` is `odd` then the next `n` of the sequence `= (3 * n) + 1`

The (unproven) [Collatz conjecture](<https://en.wikipedia.org/wiki/Collatz conjecture> "wp: Collatz conjecture") is that the hailstone sequence for any starting number always terminates.

The hailstone sequence is also known as hailstone numbers (because the values are usually subject to multiple descents and ascents like hailstones in a cloud), or as the Collatz sequence.

# --instructions--

1. Create a routine to generate the hailstone sequence for a number

> 2. Show the number less than `limit` which has the longest hailstone sequence together with that sequence's length. (But don't show the actual sequence!)

Reviewer: ```suggestion
2. `Your function should return an array with the number less than `limit` which has the longest hailstone sequence and that sequence's length. (But don't show the actual sequence!)
```

Wondering if this instruction could be a little more clear. Without looking at the tests, you cannot tell what is supposed to be returned - not sure if my suggestion is the best wording. The original has the same issue, so this problem wasn't introduced with this PR and doesn't have to be fixed since it was preexisting.
